python/paddle/fluid/optimizer.py

```diff
@@ -67,7 +67,8 @@ def __init__(self,
                  regularization=None,
                  grad_clip=None,
                  name=None):
-        self._parameter_list = parameter_list
+        self._parameter_list = list(
```

Comment:
If `self._parameter_list = list(parameter_list)`, the type of param `parameter_list` of Optimizer can change from `list` to `iterable`.